Tool/CompareAccessPolicyToRBAC.ps1

```diff
@@ -75,7 +83,20 @@ foreach($ra in $roleAssignments)
         #if all add all Key Vault RBAC data actions to per identity map
         if($ada -eq "*" -Or $ada -eq "microsoft.keyVault/*" -Or $ada  -eq "microsoft.keyVault/vaults/*")
         {
-            $RBACDataActionsByIdentity.Item($ra.ObjectId) += $AllRBACDataActions
+            $raObjectId = $ra.ObjectId
```

Comment:
It does not seem necessary to loop individually in all objects. 
In this case since all data actions must be included, emptying collections seems enough:

  $RBACDataActionsByIdentity.Item($ra.ObjectId) = @{}
  $RBACDataActionsByIdentity.Item($ra.ObjectId) += $AllRBACDataActions